src/mdsynthesis/treants.py

```diff
@@ -71,57 +71,31 @@ def universe(self):
         the universe definition entirely.
 
         """
-        # TODO: include check for changes to universe definition, not just
-        # definition absence
-        if self._universe:
-            return self._universe
-        else:
-            self.universedef._activate()
-            return self._universe
+        _args = self.universedef._args
+        if _args != self._args:
+            self._args = _args
+            kwargs = self.universedef.kwargs
+            if _args is None:
+                self._universe = None
+            else:
+                self._universe = mda.Universe(*_args, **kwargs)
+        return self._universe
 
     @universe.setter
     def universe(self, universe):
-        if universe is None:
-            self.universedef._set_topology(None)
-            self.universedef._set_trajectory([])
-            self.universedef.kwargs = None
-            self.universedef.activate()
-
-        elif not isinstance(universe, Universe):
-            raise TypeError("Cannot set to {}; must be Universe".format(
-                                type(universe)))
-        else:
-            self.universedef._set_topology(universe.filename)
-            try:  # ChainReader?
-                traj = universe.trajectory.filenames
-            except AttributeError:
-                try:  # Reader?
-                    traj = [universe.trajectory.filename]
-                except AttributeError:  # Only topology
-                    traj = []
-
-            self.universedef._set_trajectory(traj)
-            self.universedef.kwargs = universe.kwargs
-            # finally, just use this instance
-            self._universe = universe
+        self.universedef.update(universe)
+        self._universe = universe
 
     @universe.deleter
     def universe(self):
```

Comment:
what is the purpose of this delete? Can it be removed?

Reply:
It makes `del sim.universe` clear the Universe definition, but I'll admit that's a very confusing way to treat `del`. We should remove this.

Reply:
do you still want to remove this? I would to he honest I don't see the use for `del sim.universe` wouldn't `sim.universe = other_universe` make more sense for a user?

Reply:
``sim.universe = None`` would be the approach we should encourage for unsetting the universe definition. Using `del` here is just a very confusing usage, outside of what one expects `del` to actually do in Python.